src/components/Home/Highlights/PaginationCarousel.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { FC } from "react"
+import Animated, {
+    Extrapolate,
+    interpolateColors,
+} from "react-native-reanimated"
+import { View, Dimensions } from "react-native"
+import { CarouselItem } from "./HighlightsManager"
+import { usePalette } from "utils/colors"
+
+const { width } = Dimensions.get("window")
+
+/**
+ * Custom Pagination component to handle the animation of the dots
+ */
+export const PaginationCarousel: FC<{
+    dataToShow: CarouselItem[]
+    scrollX: Animated.Value<number>
+    currentIndex: number
+}> = ({ dataToShow, scrollX, currentIndex }) => {
+    const { palette } = usePalette()
+    return (
+        <View style={{ marginTop: 24 }}>
+            <Animated.FlatList
+                data={dataToShow}
+                keyExtractor={(_, idx) => idx.toString()}
+                horizontal
+                bounces={false}
+                style={{
+                    flexDirection: "row",
+                    alignSelf: "center",
+                }}
+                renderItem={({ index }) => {
+                    const inputRange = [
+                        (index - 1) * width,
+                        index * width,
+                        (index + 1) * width,
+                    ]
+
+                    const scaleOutputRange = [0.5, 1, 0.5]
+                    const colorOutputRange = [
+                        palette.lighter,
+                        palette.accent,
+                        palette.lighter,
+                    ]
+
+                    const dotScale = scrollX.interpolate({
+                        inputRange,
+                        outputRange: scaleOutputRange,
+                        extrapolate: Extrapolate.CLAMP,
+                    })
+
+                    return (
+                        <View
+                            style={{
+                                width: 11,
+                            }}
+                        >
+                            <Animated.View
+                                style={{
+                                    width: 11,
+                                    height: 11,
+                                    borderRadius: 7,
+                                    backgroundColor:
+                                        index === currentIndex
+                                            ? (interpolateColors(scrollX, {
+                                                  inputRange,
```

Comment:
per rendere l'interpolazione del colore ancora più smooth, potresti modificare l'input range, perché così com'è il "currentIndex" è uguale all'index solo quando sei già a metà scroll, quindi l'interpolazione parte già da metà.
potresti, solo se hai voglia, sostituire questo input range con qualcosa tipo
```suggestion
                                                  inputRange: [
                                                      (index - 0.5) * width,
                                                      index * width,
                                                      (index + 0.5) * width,
                                                  ],
```

Reply:
wow ora è perfetta!Grazie del tip